vertx-web-client/src/main/java/io/vertx/ext/web/client/cache/CacheManager.java

```diff
@@ -0,0 +1,38 @@
+package io.vertx.ext.web.client.cache;
+
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.impl.CacheInterceptor;
+
+import java.util.Optional;
+
+/**
+ * Abstracts caching of HTTP requests and responses
+ * @author Alexey Soshin
+ */
+public interface CacheManager {
+
+    /**
+     * Clears the cache completely
+     */
+    void flush();
+
+    /**
+     * Attempts to fetch value corresponding to request from cache
+     * @param cacheKey key based on HTTP request
+     * @return cached HTTP response or Optional.empty
+     */
+    Optional<HttpResponse<Object>> fetch(CacheInterceptor.CacheKey cacheKey);
```

Comment:
simply use `Object get(Object key)` and no dependency to CacheKey and Optional (null is very fine here)

Reply:
Why not to be specific? After all, this is a web client cache, not just any cache.

Reply:
then rather move the CacheKey to the cache package, CacheInterceptor is in `impl` package and `cache` package is SPI package

Reply:
`CacheKey` cannot be moved, since it relies on `HttpRequestImpl` protected fields.
That was the initial reason of exposing them as getters. 
Is there some security considerations for those fields? Why shouldn't we simply expose them?

Reply:
Used generics instead. That way we don't have reference to `CacheKey` in `.cache` package. WDYT?